sdk/search/azure-search-documents/README.md

```diff
@@ -420,6 +424,25 @@ SEARCH_ASYNC_CLIENT.search("luxury")
     });
 ```
 
+### Authenticate in a National Cloud
+
+To authenticate in a [National Cloud](https://docs.microsoft.com/azure/active-directory/develop/authentication-national-cloud), you will need to make the following additions to your client configuration:
+
+- Set the `AuthorityHost` in the credential options or via the `AZURE_AUTHORITY_HOST` environment variable
+- Set the `audience` in `SearchClientBuilder`, `SearchIndexClientBuilder`, or `SearchIndexerClientBuilder`
+
+```java readme-sample-nationalCloud
+// Create a SearchClient that will authenticate through AAD in the China national cloud.
+SearchClient searchClient = new SearchClientBuilder()
+    .endpoint(ENDPOINT)
+    .indexName(INDEX_NAME)
+    .credential(new EnvironmentCredentialBuilder()
+        .authorityHost(AzureAuthorityHosts.AZURE_CHINA)
+        .build())
+    .audience(SearchAudience.AZURE_CHINA)
```

Comment:
All previous examples in the readme use AzureKeyCredential. This sample uses `TokenCredential`. Also, there's a line above that says "The api-key is the sole mechanism for authenticating access to your search service endpoint.". So, that needs to be updated as well. Also, can the sample just use `DefaultAzureCredentialBuilder().authorityHost()` instead of `EnvironmentCredentialBuilder`?

Reply:
I'll update to DefaultAzureCredential.

And it looks like the statement about API key is universal across the languages, so that'll need to be updated everywhere

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/README.md#authenticate-the-client @ShivangiReja 

https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/search/search-documents/README.md#authenticate-the-client @sarangan12 

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/search/azure-search-documents/README.md#authenticate-the-client @rakshith91